baseplate/lib/thrift_pool.py

```diff
@@ -257,6 +266,10 @@ def connection(self) -> Generator[TProtocolBase, None, None]:
             # (expected) errors which should be safe for the connection.
             # don't close the transport here!
             raise
+        except self._PoolCheckoutException:
```

Comment:
Won't this swallow something like a `ServerTimeoutError`? That'd effectively make timeouts that happen during this phase never affect that request? One option would be to unwrap the original exception from the PoolCheckoutException. Another would be to decompose the two things and handle their error cases separately.

Reply:
👍